cmd/argocd/commands/plugin_test.go

```diff
@@ -0,0 +1,188 @@
+		validPrefixes:      []string{"argocd"},
+		executedPluginPath: "testdata/argocd-version",
+	}
+	o := ArgoCDCLIOptions{
+		PluginHandler: pluginsHandler,
+		Arguments:     []string{"argocd", "version", "--short", "--client"},
+	}
+	cmd.SetArgs(o.Arguments[1:])
+	err := cmd.Execute()
+	err = HandleCommandExecutionError(err, true, o)
+	output := buf.String()
+	assert.Equal(t, "argocd: v99.99.99+unknown\n", output)
+	require.NoError(t, err)
+}
+
+// Test_ArgoCD_Plugin_Successful_Execution tests for the successful execution of a plugin found in the plugin directory
+func Test_ArgoCD_Plugin_Successful_Execution(t *testing.T) {
+	cmd := NewCommand()
+	cmd.SilenceErrors = true
+	cmd.SilenceUsage = true
+
+	pluginsHandler := &testPluginHandler{
+		pluginsDirectory:   "testdata",
+		validPrefixes:      []string{"argocd"},
+		executedPluginPath: "testdata/argocd-foo",
+	}
+	o := ArgoCDCLIOptions{
+		PluginHandler: pluginsHandler,
+		Arguments:     []string{"argocd", "foo"},
+	}
+	cmd.SetArgs(o.Arguments[1:])
+
+	err := cmd.Execute()
+	require.Error(t, err, "unknown command \"foo\" for \"argocd\"")
+
+	err = HandleCommandExecutionError(err, true, o)
+	require.NoError(t, err)
+}
+
+// Test_CommandIsNeitherNormalCommandNorExistsAsPlugin checks when a command is neither a normal Argo CD CLI command nor Plugin
+func Test_CommandIsNeitherNormalCommandNorExistsAsPlugin(t *testing.T) {
+	buf := new(bytes.Buffer)
+	logrus.SetOutput(buf)
+	defer func() {
+		logrus.SetOutput(os.Stderr)
+	}()
+
+	cmd := NewCommand()
+	cmd.SilenceErrors = true
+	cmd.SilenceUsage = true
+
+	pluginsHandler := &testPluginHandler{
+		pluginsDirectory:   "testdata",
+		validPrefixes:      []string{"argocd"},
+		executedPluginPath: "",
+	}
+
+	o := ArgoCDCLIOptions{
+		PluginHandler: pluginsHandler,
+		Arguments:     []string{"argocd", "nonexistent"},
+	}
+	cmd.SetArgs(o.Arguments[1:])
+
+	err := cmd.Execute()
+	require.Error(t, err, "unknown command \"nonexistent\" for \"argocd\"")
+
+	pluginErr := HandleCommandExecutionError(err, true, o)
+	require.Equal(t, pluginErr, err)
+
+	// check for error logs
+	logOutput := buf.String()
+	assert.Contains(t, logOutput, "Error: unknown command \\\"nonexistent\\\" for \\\"argocd\\\"\\nRun 'argocd --help' for usage.\\n")
+}
+
+func Test_CommandWithInvalidArguments(t *testing.T) {
+	buf := new(bytes.Buffer)
+	logrus.SetOutput(buf)
+	defer func() {
+		logrus.SetOutput(os.Stderr)
+	}()
+
+	cmd := NewCommand()
+	cmd.SilenceErrors = true
+	cmd.SilenceUsage = true
+
+	o := ArgoCDCLIOptions{
+		Arguments: []string{"argocd", "--invalid-flag"},
+	}
+	cmd.SetArgs(o.Arguments[1:])
+
+	err := cmd.Execute()
+	require.Error(t, err, "unknown flag: --invalid-flag")
+
+	pluginErr := HandleCommandExecutionError(err, true, o)
+	require.Equal(t, pluginErr, err)
+
+	// Ensure appropriate error logs
+	logOutput := buf.String()
+	assert.Contains(t, logOutput, "Error: unknown flag: --invalid-flag")
+}
+
+// TODO
+func Test_LookForPluginError(t *testing.T) {
```

Comment:
```suggestion
func Test_LookForPluginError(t *testing.T) {
```

Instead of calling the `HandleCommandExecutionError()`, should we call the `LookForPlugin()` func and test it instead? FYI, `HandleCommandExecutionError()` calls the `LookForPlugin()` internally but that would be a pain to test this specific function thoroughly. Thoughts?

Reply:
This testcase will serve the purpose to check for relative paths too.